src/database-sqlite3.cpp

```diff
@@ -162,14 +180,53 @@ void Database_SQLite3::openDatabase()
 	SQLOK(sqlite3_busy_handler(m_database, Database_SQLite3::busyHandler,
 		m_busy_handler_data), "Failed to set SQLite3 busy handler");
 
+	bool created = false;
 	if (needs_create) {
+		created = true;
 		createDatabase();
 	}
 
 	std::string query_str = std::string("PRAGMA synchronous = ")
 			 + itos(g_settings->getU16("sqlite_synchronous"));
 	SQLOK(sqlite3_exec(m_database, query_str.c_str(), NULL, NULL, NULL),
 		"Failed to modify sqlite3 synchronous mode");
+
+	std::string journal_mode = "default-wal";
+	g_settings->getNoEx("sqlite_journal_mode", journal_mode);
+	journal_mode = uppercase(journal_mode);
+	bool change_mode = true;
+	if (journal_mode.substr(0, 8) == "DEFAULT-") {
+		change_mode = false;
+		journal_mode = journal_mode.substr(8);
+	}
+	if (journal_mode != "DELETE" && journal_mode != "TRUNCATE"
+			&& journal_mode != "PERSIST" && journal_mode != "WAL")
+		throw DatabaseException(
+			std::string("Invalid value for configuration parameter"
+					"'sqlite_journal_mode': '")
+				+ (change_mode ? "" : "default-") + lowercase(journal_mode)
+				+ "', expected: [default-]{delete|truncate|persist|wal}");
```

Comment:
Omitting braces on such a large block of code is confusing, please add braces to the `if` block.